### ISSUE TITLE
RegionCoverer for Rect broken.

### ISSUE THREAD

**Issue by feuerste** [yjh0502/rust-s2#2]
Thanks for porting s2 to Rust! This is really helpful!
I just started playing around with the `RegionCoverer` using a `Rect` as input. Unfortunately I don't get the results that I expect. This is my test program:
```rust
use s2::latlng::LatLng;
use s2::rect::Rect;
use s2::region::RegionCoverer;
use s2::s1::angle::{Angle, Deg};

fn main() {
    let lat = 37.0;
    let long = 127.0;
    let ext = 0.0002;
    let level = 20;
    let level_mod = 1;
    let max_cells = 1000;

    let center = LatLng::new(Angle::from(Deg(lat)), Angle::from(Deg(long)));
    let size = LatLng::new(Angle::from(Deg(ext)), Angle::from(Deg(ext)));
    let rect = Rect::from_center_size(center, size);
    let cov = RegionCoverer {
        min_level: level,
        max_level: level,
        level_mod,
        max_cells,
    };

    let cu = cov.covering(&rect);
    for cell_id in &cu.0 {
        println!("{:?}", cell_id);
    }
}
```
which gives me:
```
1/22233121321223133021
```

The same code using the C++ library
```cpp
#include "s2/s2latlng_rect.h"
#include "s2/s2region_coverer.h"

int main() {
  const double lat = 37.0;
  const double lng = 127.0;
  const double ext = 0.0002;
  const int level = 20;
  const int level_mod = 1;
  const int max_cells = 1000;

  const S2LatLng center = S2LatLng::FromDegrees(lat, lng);
  const S2LatLng size = S2LatLng::FromDegrees(ext, ext);
  const S2LatLngRect rect = S2LatLngRect::FromCenterSize(center, size);
  S2RegionCoverer::Options options;
  options.set_min_level(level);
  options.set_max_level(level);
  options.set_level_mod(level_mod);
  options.set_max_cells(max_cells);
  S2RegionCoverer cov(options);
  const S2CellUnion cu = cov.GetCovering(rect);
  for (const S2CellId &cell_id : cu.cell_ids()) {
    std::cout << cell_id << std::endl;
  }
  return 0;
}
```

as well as the go lib
```go
package main

import (
	"fmt"

	"github.com/golang/geo/s2"
)

func main() {
	lat := 37.0
	lng := 127.0
	ext := 0.0002
	level := 20
	level_mod := 1
	max_cells := 1000

	center := s2.LatLngFromDegrees(lat, lng)
	size := s2.LatLngFromDegrees(ext, ext)
	rect := s2.RectFromCenterSize(center, size)

	coverer := s2.RegionCoverer{
		MinLevel: level,
		MaxLevel: level,
		LevelMod: level_mod,
		MaxCells: max_cells,
	}

	cu := coverer.Covering(rect)
	for _, cell_id := range cu {
		fmt.Printf("%v\n", cell_id)
	}
}
```

give me back
```
1/22233121321223133002
1/22233121321223133012
1/22233121321223133013
1/22233121321223133020
1/22233121321223133021
1/22233121321223133022
1/22233121321223133023
1/22233121321223133031
1/22233121321223133310
1/22233121321223133311
1/22233121321223133312
1/22233121321223133313
1/22233121321223133321
```

Do you have a hunch on where the problem could be?
Thanks a lot for your help!

**Comment by feuerste** [comment]
@yjh0502 are you still following this repo?

**Comment by yjh0502** [comment]
@feuerste Sorry for late. I'll look into it.

**Comment by yjh0502** [comment]
@feuerste There was a stupid bug on `Rect::cap_bound`, https://github.com/yjh0502/rust-s2/commit/d60d6d86a324a06ad13804611def05b1b8ce4be4

Although it returns four more cells compared to reference implementations, at least the results is logically correct now.
```diff
< 1/22233121321223133132
< 1/22233121321223133133
< 1/22233121321223133200
< 1/22233121321223133201
```

**Comment by yjh0502** [comment]
I added missing `Region::intersects_cell` for `Rect`. Now rust version should work same with reference implementations. I published `0.0.8` with the fix. BTW thanks for the report!

**Comment by feuerste** [comment]
Thanks a lot for looking into and fixing this! I will test it tomorrow.
